class.jetpack-client.php

```diff
@@ -65,10 +65,18 @@ public static function remote_request( $args, $body = null ) {
 		if ( is_null( $body ) ) {
 			$body_hash = '';
 		} else {
-			if ( !is_string( $body ) ) {
+
+			// Allow arrays to be used in passing data.
+			$body_to_hash = $body;
+			if ( is_array( $body ) ) {
+				$body_to_hash = json_encode( self::_stringify_data( $body ) );
```

Comment:
Wouldn't it be simpler to just `serialize()` ?  Is there some risk of an `unserialize()` injection attack?